extensions/markdown-language-features/src/commands/openImage.ts

```diff
@@ -0,0 +1,22 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { Command } from '../commandManager';
+import { MarkdownPreviewManager } from '../preview/previewManager';
+
+export class OpenImageCommand implements Command {
+	public readonly id = '_markdown.openImage';
+
+	public constructor(
+		private readonly _webviewManager: MarkdownPreviewManager,
+	) { }
+
+	public execute(args: { resource: string, imageSource: string }) {
+		const source = vscode.Uri.parse(args.resource);
+		const { fsPath } = vscode.Uri.parse(args.imageSource);
```

Comment:
Keep this as a uri. Otherwise it will only work for `file:` images

Reply:
I was thinking the user can only open local images so It should have `fsPath`. What is other possible options?

Reply:
Local in this case would mean part of the workspace, but the workspace could be a virtual file system or a remote machine. Just using `fsPath` won't work in these cases